src/lib/services/addon-service.ts

```diff
@@ -12,13 +12,18 @@ import { IAddon, IAddonDto, IAddonStore } from '../types/stores/addon-store';
 import { IUnleashStores, IUnleashConfig } from '../types';
 import { IAddonDefinition } from '../types/model';
 import { minutesToMilliseconds } from 'date-fns';
+import { BadDataError } from '../../lib/error';
 
 const SUPPORTED_EVENTS = Object.keys(events).map((k) => events[k]);
 
 const MASKED_VALUE = '*****';
 
 const WILDCARD_OPTION = '*';
 
+// The *actual* shape of the data we receive. The description field can't be
+// enforced now without it being breaking change.
+type ActualAddonDto = Omit<IAddonDto, 'description'> & { description?: string };
```

Comment:
The type `IAddonDto` expects `description` to be required, but it is *not*. There's probably better ways to handle this, but changing the type itself had some far-reaching issues, so this is a patch for now. Suggestions are welcome.